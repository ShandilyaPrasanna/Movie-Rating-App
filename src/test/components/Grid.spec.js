import React from "react";
import ReactDOM from "react-dom";
import toJson from "enzyme-to-json";
import { shallow, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import {
  movieGrid as Grid,
  mapStateToProps
} from "../../modules/components/Grid";

jest.mock("react-dom");

const movies = [
  {
    _id: "6f69864e-8d7d-4233-8a73-34067efe8c9a",
    title: "Black Panther",
    description:
      "After his father's death, T'Challa returns home to Wakanda to inherit his throne. However, a powerful enemy related to his family threatens to attack his nation.",
    year: "2018",
    duration: "2h 15m",
    director: "Ryan Coogler",
    genre: "Fantasy,Sci-fi",
    reviews: [{ rating: "7.3/10", provider: "IMDb" }]
  },
  {
    _id: "6f69864e-8d7d-4233-8a73-34067efe8c9a",
    title: "Black Panther",
    description:
      "After his father's death, T'Challa returns home to Wakanda to inherit his throne. However, a powerful enemy related to his family threatens to attack his nation.",
    year: "2018",
    duration: "2h 15m",
    director: "Ryan Coogler",
    genre: "Fantasy,Sci-fi",
    reviews: [
      { rating: "7.3/10", provider: "IMDb" },
      { rating: "7.3/10", provider: "IMDb" }
    ]
  }
];

configure({ adapter: new Adapter() });
describe("movieGrid Test", () => {
  let tree;
  const props = {
    movies: movies,
    classes: {}
  };
  beforeEach(() => {
    tree = shallow(<Grid {...props} />);
  });

  it("renders movieGrid ", () => {
    expect(mapStateToProps({ app: { movies } }).movies).toMatchObject(movies);
    expect(toJson(tree)).toMatchSnapshot();
  });
});
