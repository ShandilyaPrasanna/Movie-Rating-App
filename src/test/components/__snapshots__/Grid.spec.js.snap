// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`movieGrid Test renders movieGrid  1`] = `
<WithStyles(ForwardRef(Grid))
  container={true}
  spacing={2}
  style={
    Object {
      "marginTop": "20px",
    }
  }
>
  <WithStyles(ForwardRef(Grid))
    item={true}
    md={6}
    sm={6}
    xs={12}
  >
    <WithStyles(ForwardRef(Card))>
      <WithStyles(ForwardRef(CardContent))>
        <WithStyles(ForwardRef(Typography))
          component="h2"
          variant="h5"
        >
          Black Panther (2018)
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(CardContent))
          style={
            Object {
              "display": "flex",
            }
          }
        >
          <WithStyles(ForwardRef(CardContent))>
            <WithStyles(ForwardRef(Container))
              style={
                Object {
                  "display": "flex",
                }
              }
            >
              <WithStyles(ForwardRef(Typography))
                component="h3"
                variant="h8"
              >
                Genre:
              </WithStyles(ForwardRef(Typography))>
              <WithStyles(ForwardRef(Typography))
                color="primary"
                component="p"
                style={
                  Object {
                    "padding": "3px",
                  }
                }
                variant="body2"
              >
                Fantasy,Sci-fi
              </WithStyles(ForwardRef(Typography))>
            </WithStyles(ForwardRef(Container))>
            <WithStyles(ForwardRef(Container))
              style={
                Object {
                  "display": "flex",
                }
              }
            >
              <WithStyles(ForwardRef(Typography))
                component="h3"
                variant="h8"
              >
                Duration:
              </WithStyles(ForwardRef(Typography))>
              <WithStyles(ForwardRef(Typography))
                color="primary"
                component="p"
                style={
                  Object {
                    "padding": "3px",
                  }
                }
                variant="body2"
              >
                2h 15m
              </WithStyles(ForwardRef(Typography))>
            </WithStyles(ForwardRef(Container))>
            <WithStyles(ForwardRef(Container))
              style={
                Object {
                  "display": "flex",
                }
              }
            >
              <WithStyles(ForwardRef(Typography))
                component="h3"
                variant="h8"
              >
                Director:
              </WithStyles(ForwardRef(Typography))>
              <WithStyles(ForwardRef(Typography))
                color="primary"
                component="p"
                style={
                  Object {
                    "padding": "3px",
                  }
                }
                variant="body2"
              >
                Ryan Coogler
              </WithStyles(ForwardRef(Typography))>
            </WithStyles(ForwardRef(Container))>
          </WithStyles(ForwardRef(CardContent))>
          <WithStyles(ForwardRef(CardMedia))
            alt="Contemplative Reptile"
            component="img"
            height="140"
            image="clapperboard-icon.jpg"
            title="Watch Movie"
          />
        </WithStyles(ForwardRef(CardContent))>
      </WithStyles(ForwardRef(CardContent))>
      <WithStyles(ForwardRef(CardContent))>
        <WithStyles(ForwardRef(Typography))
          color="textSecondary"
          component="p"
          variant="body2"
        >
          After his father's death, T'Challa returns home to Wakanda to inherit his throne. However, a powerful enemy related to his family threatens to attack his nation.
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(CardContent))>
      <WithStyles(ForwardRef(CardActions))>
        <WithStyles(ForwardRef(Typography))
          color="primary"
          component="p"
          style={
            Object {
              "padding": "3px",
            }
          }
          variant="body2"
        >
          7.3/10
        </WithStyles(ForwardRef(Typography))>
        <GradeIcon />
        <WithStyles(ForwardRef(Typography))
          color="primary"
          component="p"
          variant="body2"
        >
          IMDb
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(CardActions))>
    </WithStyles(ForwardRef(Card))>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    item={true}
    md={6}
    sm={6}
    xs={12}
  >
    <WithStyles(ForwardRef(Card))>
      <WithStyles(ForwardRef(CardContent))>
        <WithStyles(ForwardRef(Typography))
          component="h2"
          variant="h5"
        >
          Black Panther (2018)
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(CardContent))
          style={
            Object {
              "display": "flex",
            }
          }
        >
          <WithStyles(ForwardRef(CardContent))>
            <WithStyles(ForwardRef(Container))
              style={
                Object {
                  "display": "flex",
                }
              }
            >
              <WithStyles(ForwardRef(Typography))
                component="h3"
                variant="h8"
              >
                Genre:
              </WithStyles(ForwardRef(Typography))>
              <WithStyles(ForwardRef(Typography))
                color="primary"
                component="p"
                style={
                  Object {
                    "padding": "3px",
                  }
                }
                variant="body2"
              >
                Fantasy,Sci-fi
              </WithStyles(ForwardRef(Typography))>
            </WithStyles(ForwardRef(Container))>
            <WithStyles(ForwardRef(Container))
              style={
                Object {
                  "display": "flex",
                }
              }
            >
              <WithStyles(ForwardRef(Typography))
                component="h3"
                variant="h8"
              >
                Duration:
              </WithStyles(ForwardRef(Typography))>
              <WithStyles(ForwardRef(Typography))
                color="primary"
                component="p"
                style={
                  Object {
                    "padding": "3px",
                  }
                }
                variant="body2"
              >
                2h 15m
              </WithStyles(ForwardRef(Typography))>
            </WithStyles(ForwardRef(Container))>
            <WithStyles(ForwardRef(Container))
              style={
                Object {
                  "display": "flex",
                }
              }
            >
              <WithStyles(ForwardRef(Typography))
                component="h3"
                variant="h8"
              >
                Director:
              </WithStyles(ForwardRef(Typography))>
              <WithStyles(ForwardRef(Typography))
                color="primary"
                component="p"
                style={
                  Object {
                    "padding": "3px",
                  }
                }
                variant="body2"
              >
                Ryan Coogler
              </WithStyles(ForwardRef(Typography))>
            </WithStyles(ForwardRef(Container))>
          </WithStyles(ForwardRef(CardContent))>
          <WithStyles(ForwardRef(CardMedia))
            alt="Contemplative Reptile"
            component="img"
            height="140"
            image="clapperboard-icon.jpg"
            title="Watch Movie"
          />
        </WithStyles(ForwardRef(CardContent))>
      </WithStyles(ForwardRef(CardContent))>
      <WithStyles(ForwardRef(CardContent))>
        <WithStyles(ForwardRef(Typography))
          color="textSecondary"
          component="p"
          variant="body2"
        >
          After his father's death, T'Challa returns home to Wakanda to inherit his throne. However, a powerful enemy related to his family threatens to attack his nation.
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(CardContent))>
      <WithStyles(ForwardRef(CardActions))>
        <WithStyles(ForwardRef(Typography))
          color="primary"
          component="p"
          style={
            Object {
              "padding": "3px",
            }
          }
          variant="body2"
        >
          7.3/10
        </WithStyles(ForwardRef(Typography))>
        <GradeIcon />
        <WithStyles(ForwardRef(Typography))
          color="primary"
          component="p"
          variant="body2"
        >
          IMDb
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          color="primary"
          component="p"
          style={
            Object {
              "marginLeft": "auto",
              "padding": "3px",
            }
          }
          variant="body2"
        >
          7.3/10
        </WithStyles(ForwardRef(Typography))>
        <GradeIcon />
        <WithStyles(ForwardRef(Typography))
          color="primary"
          component="p"
          variant="body2"
        >
          IMDb
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(CardActions))>
    </WithStyles(ForwardRef(Card))>
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Grid))>
`;
